advanced_training_patterns.py

```
@@ -0,0 +1,326 @@
+        target = torch.randint(0, 2, (32,))
+        adaptive_training_step(model, data, target, optimizer, use_gradients=(i % 2 == 0))
+
+
+def compute_ensemble_loss(auxiliary_model, data, target):
+    """Compute loss from auxiliary model"""
+    output = auxiliary_model(data)
+    loss = nn.functional.cross_entropy(output, target)
+    return loss
+
+
+def train_primary_with_ensemble(primary_model, auxiliary_model, optimizer, data, target):
```

Ruff (0.13.1)

Unused function argument: `primary_model`

(ARG001)